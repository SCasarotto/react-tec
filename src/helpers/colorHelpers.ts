export const componentToHex = (component: number) => {
  // @ts-ignore
  const x = component.toString(16);
  return x.length === 1 ? '0' + x : x;
};
export const hexToRgb = (hex: string) => {
  // Expand shorthand form (e.g. "03F") to full form (e.g. "0033FF")
  const shorthandRegex = /^#?([a-f\d])([a-f\d])([a-f\d])$/i;
  hex = hex.replace(shorthandRegex, function (_, r, g, b) {
    return r + r + g + g + b + b;
  });

  const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
  return result
    ? {
        r: parseInt(result[1], 16),
        g: parseInt(result[2], 16),
        b: parseInt(result[3], 16),
      }
    : null;
};
export const colorBetweenColors = (
  color1: string,
  color2: string,
  percBetween: number,
) => {
  const color1RGB = hexToRgb(color1);
  const color2RGB = hexToRgb(color2);

  if (!color1RGB || !color2RGB) {
    throw new Error('Invalid colors supplied');
  }
  const r = Math.ceil(
    color1RGB.r * percBetween + color2RGB.r * (1 - percBetween),
  );
  const g = Math.ceil(
    color1RGB.g * percBetween + color2RGB.g * (1 - percBetween),
  );
  const b = Math.ceil(
    color1RGB.b * percBetween + color2RGB.b * (1 - percBetween),
  );

  return '#' + componentToHex(r) + componentToHex(g) + componentToHex(b);
};
