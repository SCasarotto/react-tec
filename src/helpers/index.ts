export * from './colorHelpers'
export * from './manipulateRowData'
