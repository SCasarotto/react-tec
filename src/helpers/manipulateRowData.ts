interface ButtonData {
  label: string;
  value?: string;
  key?: string;
}
interface ManipulateRowData {
  rowData: string | ButtonData;
  labelForKey: string;
}
export const manipulateRowData = ({
  rowData,
  labelForKey = '',
}: ManipulateRowData) => {
  //Allowing For Greater Shorthand
  if (typeof rowData === 'string') {
    const label = rowData;
    const value = rowData;
    const key = labelForKey + rowData;
    return { label, value, key };
  } else if (typeof rowData === 'object') {
    const { label } = rowData;
    let { value, key } = rowData;

    if (!value) {
      value = label;
    }
    if (!key) {
      key = labelForKey + label;
    }

    return { label, value, key };
  }
  return rowData;
};
