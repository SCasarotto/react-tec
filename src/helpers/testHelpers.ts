export const mockOnScroll = () => {
	const { scrollTo } = window
	beforeAll(() => {
		// Replace with custom value
		window.scrollTo = jest.fn()
	})
	afterAll(() => {
		// Restore original
		window.scrollTo = scrollTo
	})
}
