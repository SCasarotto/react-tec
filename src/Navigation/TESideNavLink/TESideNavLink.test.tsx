import React from 'react'
import ReactDOM from 'react-dom'
import { TESideNavLink, TEAppWrapper } from '../..'
import { mockOnScroll } from '../../helpers/testHelpers'

describe('TESideNavLink', () => {
	mockOnScroll()
	it('renders without crashing', () => {
		const div = document.createElement('div')
		ReactDOM.render(
			<TEAppWrapper>
				<TESideNavLink title='MyTitle' to='/SomeURL' />
			</TEAppWrapper>,
			div,
		)
		ReactDOM.unmountComponentAtNode(div)
	})
})
