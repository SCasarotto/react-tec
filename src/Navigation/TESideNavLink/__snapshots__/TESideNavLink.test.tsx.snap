// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TESideNavLink matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="sc-bdnylx jOVwxu TESideNavLink "
  >
    <a
      class="sc-gtssRu fKzDzg TESideNavLinkLink"
      href="/SomeURL"
    >
      <span
        class="sc-dlnjPT iLyLvX TESideNavLinkTitle"
      >
        MyTitle
      </span>
    </a>
  </li>
  <div
    class="sc-hKFyIo dfSMMW TEPopup sc-gKAblj bsOgow TEAlert "
  >
    <div
      class="sc-eCApGN fnoCH TEPopupContent"
    >
      <button
        class="sc-jSFkmK iZcXEK TEButton sc-pNWxx gjLRwR TEAlertButton"
        type="button"
      >
        Okay
      </button>
    </div>
  </div>
  <div
    class="sc-hKFyIo dfSMMW TEPopup sc-jrsJCI YHRny TEConfirm "
  >
    <div
      class="sc-eCApGN fnoCH TEPopupContent"
    >
      <div
        class="sc-crzoUp lbwhup TEConfirmButtonoContainer"
      >
        <button
          class="sc-jSFkmK iZcXEK TEButton sc-dIsAE gIEwOq TEConfirmButton TEConfirmLeftButton"
          type="button"
        >
          Cancel
        </button>
        <button
          class="sc-jSFkmK iZcXEK TEButton sc-bqGHjH deQZlO TEConfirmButton TEConfirmRightButton"
          type="button"
        />
      </div>
    </div>
  </div>
  <div
    class="sc-hKFyIo dfSMMW TEPopup sc-fFSRdu eGuebw TENetworkActivity "
  >
    <div
      class="sc-eCApGN fnoCH TEPopupContent"
    >
      <div
        class="sc-ksluoS hlkcIu TESpinner sc-bkbjAj cggGyk TENetworkActivitySpinner"
      >
        <div
          class="sc-hBMVcZ kOddsC TESpinnerContainer"
        >
          <div
            class="sc-fnVYJo giXuWa TESpinnerRing TESpinnerRing1"
          />
          <div
            class="sc-fnVYJo gPpyAp TESpinnerRing TESpinnerRing2"
          />
          <div
            class="sc-fnVYJo ipFULs TESpinnerRing TESpinnerRing3"
          />
          <div
            class="sc-fnVYJo dpTdCu TESpinnerRing TESpinnerRing4"
          />
          <div
            class="sc-fnVYJo ggwrpK TESpinnerRing TESpinnerRing5"
          />
          <div
            class="sc-fnVYJo cwqgSW TESpinnerRing TESpinnerRing6"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
